core/src/main/scala/chisel3/Bits.scala

```diff
@@ -876,10 +905,10 @@ sealed class SInt private[chisel3] (width: Width) extends Bits(width) with Num[S
   /** @group SourceInfoTransformMacro */
   def do_=== (that: SInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = compop(sourceInfo, EqualOp, that)
 
-//  final def abs(): UInt = macro SourceInfoTransform.noArg
+//  final def abs(): UInt = macro SourceInfoTransform.noArgDummy
 
   def do_abs(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): SInt = {
-    Mux(this < 0.S, (-this), this)
+    Mux(this < 0.S, (this.unary_-(sourceInfo, compileOptions)), this)
```

Comment:
Why is this change necessary? Do you get deprecation warnings any time you do `-x` now??

Reply:
The compiler was complaining about ambiguity between this and the dummy form.

Reply:
Right, so that ambiguity is a huge problem because it basically means users can't use `-` in a unary way any more on `SInts`. This should've been caught by the tests but apparently we never finished this particular test (https://github.com/chipsalliance/chisel3/blob/958904cb2f2f65d02b2ab3ec6d9ec2e06d04e482/src/test/scala/chiselTests/SIntOps.scala#L47).

Please make that test be active and we need to find a way to not have the ambiguity because obviously `-mySInt` needs to work. Do you know why this same issue doesn't show up for stuff like `unary_!` on `Bool`? I tried it out locally, confirmed this issue but was not able to see the same issue on `Bool` which is very weird.

Reply:
@jared-barocsi did you add the test to be active? Seems like we still have not gotten to the bottom of this question?

Reply:
@mwachs5 Yes, I just now fixed the ambiguity issue